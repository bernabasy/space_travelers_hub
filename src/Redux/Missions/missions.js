import { createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const missionsAPI = 'https://api.spacexdata.com/v3/missions';

const initialState = [];
const FETCHED_MISSIONS = 'space_travelers_hub/missions/FETCHED_MISSIONS';

const fetchMissions = createAsyncThunk(
  FETCHED_MISSIONS,
  async () => {
    const response = await axios.get(missionsAPI);
    const { data } = response;
    const missions = data.map((mission) => ({
      name: mission.mission_name,
      id: mission.mission_id,
      description: mission.description,
      membership: true,
    }));
    return missions;
  },
);

const missionsReducer = (state = initialState, action) => {
  switch (action.type) {
    case `${FETCHED_MISSIONS}/fulfilled`:
      return [...state, ...action.payload];
    default:
      return state;
  }
};

export { fetchMissions };
export default missionsReducer;
