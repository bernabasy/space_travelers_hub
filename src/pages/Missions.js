import { MDBTable, MDBTableHead, MDBTableBody } from 'mdb-react-ui-kit';
import Mission from '../components/Mission';

const Missions = () => (
  <section className="m-5 d-flex justify-content-around">
    <MDBTable bordered striped align="middle" className="w-auto">
      <MDBTableHead>
        <tr>
          <th scope="col">Mission</th>
          <th scope="col">Description</th>
          <th scope="col">Status</th>
          <th scope="col"> </th>
        </tr>
      </MDBTableHead>
      <MDBTableBody>
        <Mission />
        <Mission />
        <Mission />
      </MDBTableBody>
    </MDBTable>
  </section>
);

export default Missions;
