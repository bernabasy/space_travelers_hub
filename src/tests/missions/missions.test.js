import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { render, screen, cleanup } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import Missions from '../../pages/Missions';
import missionsReducer from '../../Redux/Missions/missions';

const initialMissionState = {
  missions: [
    {
      id: '9D1B7E0',
      name: 'Thaicom',
      description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
      joined: false,
    },
  ],
};

const store = configureStore({
  reducer: missionsReducer,
  preloadedState: initialMissionState,
});

describe('Missions', () => {
  afterEach(cleanup);

  it('Missions page should render correctly', () => {
    const mission = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(mission).toMatchSnapshot();
  });

  it('Mission page should display', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const missionName = screen.getByText('Thaicom');
    expect(missionName).toBeInTheDocument();
  });
});
