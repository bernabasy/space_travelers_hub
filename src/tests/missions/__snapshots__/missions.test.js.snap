// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Mission should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="m-5 d-flex justify-content-around"
      >
        <table
          class="table align-middle table-bordered table-striped w-auto"
        >
          <thead
            class=""
          >
            <tr>
              <th
                scope="col"
              >
                Mission
              </th>
              <th
                scope="col"
              >
                Description
              </th>
              <th
                scope="col"
              >
                Status
              </th>
              <th
                scope="col"
              >
                 
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr>
              <td
                class="fw-bold align-top"
              >
                Thaicom
              </td>
              <td>
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td>
                <button
                  class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                  disabled=""
                  role="button"
                >
                  NOT A MEMBER
                </button>
              </td>
              <td>
                <button
                  class="btn btn-outline-dark text-nowrap"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="m-5 d-flex justify-content-around"
    >
      <table
        class="table align-middle table-bordered table-striped w-auto"
      >
        <thead
          class=""
        >
          <tr>
            <th
              scope="col"
            >
              Mission
            </th>
            <th
              scope="col"
            >
              Description
            </th>
            <th
              scope="col"
            >
              Status
            </th>
            <th
              scope="col"
            >
               
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr>
            <td
              class="fw-bold align-top"
            >
              Thaicom
            </td>
            <td>
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td>
              <button
                class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                disabled=""
                role="button"
              >
                NOT A MEMBER
              </button>
            </td>
            <td>
              <button
                class="btn btn-outline-dark text-nowrap"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Mission Missions should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="m-5 d-flex justify-content-around"
      >
        <table
          class="table align-middle table-bordered table-striped w-auto"
        >
          <thead
            class=""
          >
            <tr>
              <th
                scope="col"
              >
                Mission
              </th>
              <th
                scope="col"
              >
                Description
              </th>
              <th
                scope="col"
              >
                Status
              </th>
              <th
                scope="col"
              >
                 
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr>
              <td
                class="fw-bold align-top"
              >
                Thaicom
              </td>
              <td>
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td>
                <button
                  class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                  disabled=""
                  role="button"
                >
                  NOT A MEMBER
                </button>
              </td>
              <td>
                <button
                  class="btn btn-outline-dark text-nowrap"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="m-5 d-flex justify-content-around"
    >
      <table
        class="table align-middle table-bordered table-striped w-auto"
      >
        <thead
          class=""
        >
          <tr>
            <th
              scope="col"
            >
              Mission
            </th>
            <th
              scope="col"
            >
              Description
            </th>
            <th
              scope="col"
            >
              Status
            </th>
            <th
              scope="col"
            >
               
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr>
            <td
              class="fw-bold align-top"
            >
              Thaicom
            </td>
            <td>
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td>
              <button
                class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                disabled=""
                role="button"
              >
                NOT A MEMBER
              </button>
            </td>
            <td>
              <button
                class="btn btn-outline-dark text-nowrap"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Missions Missions page should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="m-5 d-flex justify-content-around"
      >
        <table
          class="table align-middle table-bordered table-striped w-auto"
        >
          <thead
            class=""
          >
            <tr>
              <th
                scope="col"
              >
                Mission
              </th>
              <th
                scope="col"
              >
                Description
              </th>
              <th
                scope="col"
              >
                Status
              </th>
              <th
                scope="col"
              >
                 
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr>
              <td
                class="fw-bold align-top"
              >
                Thaicom
              </td>
              <td>
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td>
                <button
                  class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                  disabled=""
                  role="button"
                >
                  NOT A MEMBER
                </button>
              </td>
              <td>
                <button
                  class="btn btn-outline-dark text-nowrap"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="m-5 d-flex justify-content-around"
    >
      <table
        class="table align-middle table-bordered table-striped w-auto"
      >
        <thead
          class=""
        >
          <tr>
            <th
              scope="col"
            >
              Mission
            </th>
            <th
              scope="col"
            >
              Description
            </th>
            <th
              scope="col"
            >
              Status
            </th>
            <th
              scope="col"
            >
               
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr>
            <td
              class="fw-bold align-top"
            >
              Thaicom
            </td>
            <td>
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td>
              <button
                class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                disabled=""
                role="button"
              >
                NOT A MEMBER
              </button>
            </td>
            <td>
              <button
                class="btn btn-outline-dark text-nowrap"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Missions Missions should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="m-5 d-flex justify-content-around"
      >
        <table
          class="table align-middle table-bordered table-striped w-auto"
        >
          <thead
            class=""
          >
            <tr>
              <th
                scope="col"
              >
                Mission
              </th>
              <th
                scope="col"
              >
                Description
              </th>
              <th
                scope="col"
              >
                Status
              </th>
              <th
                scope="col"
              >
                 
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr>
              <td
                class="fw-bold align-top"
              >
                Thaicom
              </td>
              <td>
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td>
                <button
                  class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                  disabled=""
                  role="button"
                >
                  NOT A MEMBER
                </button>
              </td>
              <td>
                <button
                  class="btn btn-outline-dark text-nowrap"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="m-5 d-flex justify-content-around"
    >
      <table
        class="table align-middle table-bordered table-striped w-auto"
      >
        <thead
          class=""
        >
          <tr>
            <th
              scope="col"
            >
              Mission
            </th>
            <th
              scope="col"
            >
              Description
            </th>
            <th
              scope="col"
            >
              Status
            </th>
            <th
              scope="col"
            >
               
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr>
            <td
              class="fw-bold align-top"
            >
              Thaicom
            </td>
            <td>
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td>
              <button
                class="ripple ripple-surface btn btn-dark btn-sm text-nowrap"
                disabled=""
                role="button"
              >
                NOT A MEMBER
              </button>
            </td>
            <td>
              <button
                class="btn btn-outline-dark text-nowrap"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
